solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -573,6 +576,25 @@ public AssignRequest build() {
         }
       return result;
     }
+
+    private void filterDedicatedOverseers(SolrCloudManager solrCloudManager, List<String> nodeList) {
+      try {
+        if(solrCloudManager.getDistribStateManager().hasData(ZkStateReader.ROLES)) {
+          VersionedData data = solrCloudManager.getDistribStateManager().getData(ZkStateReader.ROLES);
+          Map<String, Object> roles = (Map<String, Object>) Utils.fromJSON(data.getData());
+          List<String> dedicatedOverseers = (List<String>) roles.getOrDefault("overseer", Collections.emptyList());
+          for (String dedicatedOverseer : dedicatedOverseers) {
+            if(nodeList.contains(dedicatedOverseer)) {
+              nodeList.remove(dedicatedOverseer);
+            }
+          }
+        }
+
+      } catch (Exception e) {
```

Comment:
@noblepaul @chatman I am not quite comfortable on eating the exception here and that other place above.  We have big clusters and things happen continuously(shard move, shard spilt, gc, zk connect/d) and we may run into this some day. Would be good to have some retry or fallback on this!